Corona-Warn-App/src/test/java/de/rki/coronawarnapp/ccl/dccwalletinfo/calculation/DccWalletInfoCalculationTest.kt

```diff
@@ -1,35 +1,127 @@
+import com.fasterxml.jackson.databind.node.NullNode
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.fasterxml.jackson.databind.node.TextNode
+import de.rki.coronawarnapp.ccl.configuration.model.CCLConfiguration
+import de.rki.coronawarnapp.ccl.configuration.model.FunctionDefinition
+import de.rki.coronawarnapp.ccl.configuration.model.FunctionParameter
+import de.rki.coronawarnapp.ccl.configuration.model.JsonFunctionsDescriptor
+import de.rki.coronawarnapp.ccl.configuration.storage.CCLConfigurationRepository
 import de.rki.coronawarnapp.ccl.dccwalletinfo.model.CclCertificate
 import de.rki.coronawarnapp.ccl.dccwalletinfo.model.Cose
 import de.rki.coronawarnapp.ccl.dccwalletinfo.model.Cwt
+import de.rki.coronawarnapp.ccl.dccwalletinfo.model.DccWalletInfo
 import de.rki.coronawarnapp.covidcertificate.common.certificate.CwaCovidCertificate
 import de.rki.coronawarnapp.covidcertificate.common.certificate.DccData
 import de.rki.coronawarnapp.covidcertificate.common.certificate.VaccinationDccV1
 import de.rki.coronawarnapp.util.qrcode.coil.CoilQrCode
 import de.rki.coronawarnapp.util.serialization.SerializationModule
 import io.kotest.matchers.shouldBe
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
 import io.mockk.every
+import io.mockk.impl.annotations.MockK
 import io.mockk.mockk
 import org.joda.time.DateTime
 import org.joda.time.Instant
+import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import testhelpers.BaseTest
 
 class DccWalletInfoCalculationTest : BaseTest() {
 
-    private val instance = DccWalletInfoCalculation(
-        gson = SerializationModule().baseGson(),
-        mapper = SerializationModule().jacksonObjectMapper()
+    @MockK lateinit var cclConfigurationRepository: CCLConfigurationRepository
+    @MockK lateinit var walletInfo: DccWalletInfo
+    @MockK lateinit var cclJsonFunctions: CclJsonFunctions
+    @MockK lateinit var mapper: ObjectMapper
+
+    val param1 = FunctionParameter(
+        name = "greeting",
+        default = TextNode.valueOf("Hello")
+    )
+
+    val param2 = FunctionParameter(
+        name = "name"
+    )
+
+    val jfnLogic = ObjectMapper().readTree(
+        """{"return": [{"concatenate": [{"var": "greeting"}," ",{"var": "name"}]}]}"""
+    )
+
+    val jfnDescriptor = JsonFunctionsDescriptor(
+        name = "greet",
+        definition = FunctionDefinition(
+            parameters = listOf(param1, param2),
+            logic = listOf(jfnLogic)
+        )
     )
 
+    private val config = CCLConfiguration(
+        identifier = "CCL-DE-0001",
+        type = CCLConfiguration.Type.CCLConfiguration,
+        country = "DE",
+        version = "1.0.0",
+        schemaVersion = "1.0.0",
+        engine = "JsonFunctions",
+        engineVersion = "1.0.0",
+        _validFrom = "2021-10-07T00:00:00Z",
+        _validTo = "2030-06-01T00:00:00Z",
+        logic = CCLConfiguration.Logic(jfnDescriptors = listOf(jfnDescriptor))
+    )
```

Comment:
This is not needed for this test, I think? Also the `cclConfigurationRepository`

Reply:
why ? I need to provide a list of configurations

Reply:
Ah, I see this is another test :D